### ISSUE TITLE
Carla not loading in LMMS nightly

### ISSUE THREAD

**Issue by kamkamkamkamkamkamkamkam** [LMMS/lmms#6984]
# Please search the issue tracker for existing bug reports before submitting your own. Delete this line to confirm no similar report has been posted yet.

### Bug Summary
Carla fails to load in the latest nightly
#### Steps to reproduce
Install Carla on Linux system. Download LMMS Nightly build. Run LMMS.
#### Expected behavior
Carla appears in instrument list.
#### Actual behavior
Carla fails to appear in instrument list.
#### Screenshot
![image](https://github.com/LMMS/lmms/assets/65249472/a3b6208c-b460-43a5-be5a-fa95e6622c1c)

#### Affected LMMS versions
Nightly

#### Logs
```
[kam@fedora ~]$ cd Downloads/
[kam@fedora Downloads]$ ./2023-11-11-lmms-1.3.0-alpha.1.470+gecc5ff8ca-linux-x86_64.AppImage 
Carla appears to be installed on this system at /usr/lib[64]/carla so we'll use it.
Jack appears to be installed on this system, so we'll use it.
Lv2 plugin SUMMARY: 12 of 13  loaded in 90 msecs.
For details about not loaded plugins, please set
  environment variable "LMMS_LV2_DEBUG" to nonempty.
Lv2 Plugins blacklisted: 1 of 13 
  If you want to ignore the blacklist (dangerous!), please set
  environment variable "LMMS_IGNORE_BLACKLIST" to nonempty.
Cannot load library /tmp/.mount_2023-1JyYgPw/usr/lib/lmms/libcarlapatchbay.so: (/lib64/libgobject-2.0.so.0: undefined symbol: g_uri_ref)
Cannot load library /tmp/.mount_2023-1JyYgPw/usr/lib/lmms/libcarlabase.so: (/lib64/libgobject-2.0.so.0: undefined symbol: g_uri_ref)
Cannot load library /tmp/.mount_2023-1JyYgPw/usr/lib/lmms/libcarlarack.so: (/lib64/libgobject-2.0.so.0: undefined symbol: g_uri_ref)
```

**Comment by djent** [comment]
I was able to fix this by deleting by deleting libglib-2.0.so.0 from the AppImage
  
#### Extract appimage
```
./lmms-1.3.0-alpha.1.481+g3a928d80b-linux-x86_64.AppImage --appimage-extract
```
  
This will create a directory called `squashfs-root`
  
delete `libgobject-2.0.so.0` from `squashfs-root/usr/lib`  
  
#### See if it works
```
# cd into squashfs-root and run AppRun  
./AppRun
```
  
If you want to create a new AppImage you will need to use `appimagetool`.

 cd to one directory above squashfs-root and run the following:
```ARCH=x86_64 appimagetool squashfs-root```

Hope this helps

**Comment by luzpaz** [comment]
Can that appimage get rebuilt ?

**Comment by HazySkies** [comment]
> I was able to fix this by deleting by deleting libglib-2.0.so.0 from the AppImage
Thanks a bunch! I was having the same issue as OP and this was all I needed to do to solve my issue.

The rest of the instructions were kinda hard to follow, because strangely so such `libgobject-2.0.so.0` exists within `squashfs-root/usr/lib`, Which might be because I'm using a newer nightly though (`lmms-1.3.0-alpha.1.490+g6b21dc789-linux-x86_64.AppImage`).

Either way, the rest of your provided method on how to then launch the application after the quick fix was all that was necessary for me to start working with Carla again with the newest version, thanks again ^-^

**Comment by super-manis** [comment]
So what is the actual fix. Sorry if i missed something. As it has been previously mentioned the libgobject-2.0.so.0 does not exist on the appimage.

**Comment by zynskeywolf** [comment]
It's not only an issue with the appimage but with local builds too in my case. Carla does appear in the list when started directly from the build folder, and also when installed *only if* the prefix is `/usr/local`, but it does *not* show up if it's `/usr`.

**Comment by super-manis** [comment]
Got it! The latest appimage builds are optimized for Carla 2.5.8. Since i am on Ubuntu 22 the system version of Carla is much older than that so you need to update it manually. If you want me to post the steps here please reply to this post. Once Carla is up to date it will be loaded normally on the appimage builds. It still doesn't load on local builds though which is still a bit concerning...

**Comment by rdrpenguin04** [comment]
This still seems to be an issue on local builds on Kubuntu 25.10.

**Comment by zatrit** [comment]
The issue still persists on a nightly build for Arch Linux

**Comment by kamkamkamkamkamkamkamkam** [comment]
I think it's worth nothing that I haven't experienced this bug in years since I created this issue, also on Arch Linux
